### PR TITLE
Set UID/GID for ARC runner builds

### PR DESCRIPTION
**Pull request by fheinecke**
ARM64 builds are currently failing as the makefile incorrectly assumes that all CI systems we have use UID/GID 1000. This PR, along with https://github.com/gravitational/cloud-terraform/pull/2473, set the UID/GID to the value required by the new runners.

Confirmation that this value is set correctly: https://github.com/gravitational/dev-fred-gha-eks/actions/runs/5215479394/jobs/9413060326?pr=58#step:3:52

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

File: build.assets/Makefile
Comment (jakule) [inline build.assets/Makefile:75]: Won't that break the image generation? We use that value when creating images. If we change the default user then the generated images will also have a different default user, so when you run that image somewhere you will probably get a permission error.

**Comment by fheinecke** [comment]
Holding off on merging this until https://github.com/gravitational/cloud-terraform/pull/2487 lands so that I can do a test run before merge

File: build.assets/Makefile
Comment (fheinecke) [reply build.assets/Makefile:75]: Did some spot checks and this doesn't seem to affect any customer facing images. Right now it only appears to affect buildboxes, which now also expect (in the context of ARC runners) uid/gid of 1001.

**Pull request merged**
